### ISSUE TITLE
Command /baltop causes exceptions

### ISSUE THREAD

**Issue by ChrisTaylor201** [EssentialsX/Essentials#3991]
**Type of bug**

Error in console

**`/ess version` output**

Server version: 1.16.5-R0.1-SNAPSHOT git-Paper-467 (MC: 1.16.5)
EssentialsX version: 2.19.0-dev+71-f29f9e1
PlaceholderAPI version: 2.10.9
LuckPerms version: 5.2.106
Vault version: 1.7.3-b131
EssentialsXSpawn version: 2.19.0-dev+71-f29f9e1
Citizens version: 2.0.27-SNAPSHOT (build 2264)
ChestShop version: 3.11 (build 213)

**Server startup log**

https://gist.github.com/christaylor201/827a14b468680afd17b8406d91a77cc9

**EssentialsX config files**

https://gist.github.com/christaylor201/b76ea8749ee7c84cad1aa1833680c018

**Error log (if applicable)**

https://gist.github.com/christaylor201/a928d8d6a3a63adde8142d64cb82cf6f

**Bug description**

If a player runs `/baltop` command, it will result in exceptions, see the log above.

**Steps to reproduce**

1. Start an offline mode server.
2. Make sure to have at least one player with some balance.
3. Run `/baltop` command.
4. Check the console for the exceptions mentioned above.

**Expected behaviour**

Running `/baltop` command shouldn't cause any exceptions.

**Actual behaviour**

Running `/baltop` command causes exceptions.

**Additional information**

* This issue should be related to #3974 and #3978.
* We've never used any other server software than Paper.

**Comment by Krakenied** [comment]
2.19.0-dev+74 too
https://gist.github.com/MrKrakenied/5cc8990fe4f8cd38a41e1de27903e434

**Comment by Krakenied** [comment]
[11:21:51 INFO]: Server: 1.16.5-R0.1-SNAPSHOT git-Purpur-978 (MC: 1.16.5)
[11:21:51 INFO]: EssentialsX: 2.19.0-dev+74-36422ab
[11:21:51 INFO]: PlaceholderAPI: 2.10.10-DEV-122
[11:21:51 INFO]: LuckPerms: 5.2.106
[11:21:51 INFO]: Vault: 1.7.3-b131
[11:21:51 INFO]: EssentialsXSpawn: 2.19.0-dev+74-36422ab
[11:21:51 INFO]: Citizens: 2.0.27-SNAPSHOT (build 2252)
[11:21:51 INFO]: ChestShop: 3.12-SNAPSHOT (build 275)

**Comment by MrKrakenied** [comment]
another error https://gist.github.com/MrKrakenied/b458e3f43784f7fc2f3a0b263c133c89

**Comment by Krakenied** [comment]
#4008

**Comment by Bobcat00** [comment]
I'm seeing the same exception for a totally online server when players change their names.

**Comment by JRoy** [comment]
> I'm seeing the same exception for a totally online server when players change their names.

Expected, my pr for the vault adapter should fix that. Feel free to try it out and let me know of the results.

**Comment by JRoy** [comment]
Hello, this issue has been mitigated in recent EssentialsX dev builds!

If you currently have damaged data, you can read more about the steps you can take to fix this and prevent further issues of this sort [here](https://github.com/EssentialsX/Essentials/issues/3956#issuecomment-779254544).

Download the latest dev build [here](https://essentialsx.net/downloads.html).

**Comment by ChrisTaylor201** [comment]
I apologize for deleting my previous comment, but it was no longer relevant – if I had waited for an update of the Vault expansion for PlaceholderAPI, this issue wouldn't have been solved.

We had 3,086 user data files in total and 525 of them were corrupted. Thanks to the provided solution, I managed to find and delete them, thus fixing the issue. There were also 4 other corrupted files without the `npc: true` line which I had to delete.

Nevertheless, while I was searching for these files, I noticed 99% of them were created at the same time; it seemed odd to me. I searched through the server log from that day, which we luckily still had, and found out what had happened.

Over a year ago on that day, the server owner performed `/reload` command because another plugin had stopped working. There were a lot of players online, that's why he ran it, although I had always advised him not to. This action caused hundreds of corrupted files to be generated, even logging them to the console.

Later that day, we discussed it and thought it hadn't done any damage, but we were wrong.

Anyway, I want to thank you for resolving these exceptions! If you hadn't decided to log them by default, I would have thought our user data were alright.
